### PR TITLE
Allow to ignore files during diff generation

### PR DESCRIPTION
**Pull request by delvh**
Going forward, this PR allows us to ignore certain files while generating diffs.
~~While the functionality is currently unused, I intend to use it in a follow-up PR in the next days/ weeks to implement viewing files in a PR.
I could also postpone this PR and then post all changes at once in the other PR (once it's ready).~~
The "other PR" is #19007 and was implemented without this functionality.
As it is thematically isolated and decreases the size of the other PR (it will likely become large), I would like to merge this now.
Once both PRs are merged, we can hopefully see a **huge** performance boost for PRs with large files that have already been viewed by the user.

I currently know of one problem and of one potential problem that need to be answered before merging this PR:
~~1. How should the ignored files be reported? Because with the proposed approach, they will simply be missing in the output, which is understandable, but not really user-friendly. Is there an inexpensive way to get all information stored in the `DiffFile` struct from git?~~
Not a problem as I only had the use case for #19007 in mind which was now implemented without it.
2. How are the git commands executed? In particular, will the quoting (`''`) be problematic on Windows systems? I don't have a Windows system at hand right now, so it would be good if someone else could confirm this.

### CODE REVIEW

**Comment by delvh** [comment]
Well... The problem with testing at the moment is pretty much that what I mentioned in `1.` and `2.`...
I mean, I can test for its absence, but more is right now not possible.

File: services/gitdiff/gitdiff.go [hunk 1]
```diff
@@ -1327,33 +1328,44 @@ func GetDiff(gitRepo *git.Repository, opts *DiffOptions, files ...string) (*Diff
 	if len(opts.SkipTo) > 0 {
 		argsLength++
 	}
-	if len(files) > 0 {
-		argsLength += len(files) + 1
+	argsLength += len(files)
+
+	// git 1.9.0 added pathspec exclusion of files
```
Comment (lunny) [inline services/gitdiff/gitdiff.go:1333]: Now the minimal git version is 2.0, so version check is unnecessary.

Comment (delvh) [reply services/gitdiff/gitdiff.go:1333]: Yep, true.
I'd still like to add some tests before allowing this PR to be merged.
However, I'm still unsure how to do that in this case:
I know there are former git repositories in our codebase that are used for testing somehow, but I have no idea how or where to add tests.

I'll fix it when I add the tests as well.

**Comment by puni9869** [comment]
@delvh is this still active PR?

**Comment by delvh** [comment]
Not really

**Pull request closed without merging**